test/intl402/Temporal/PlainMonthDay/from/buddhist-month-codes.js

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2025 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plainmonthday.from
+description: PlainMonthDay can be created for all month codes (M01-M12) in Buddhist calendar
+features: [Temporal]
+includes: [temporalHelpers.js]
+---*/
+
+// Test that all month codes M01-M12 are valid for the Buddhist calendar
+// The Buddhist calendar follows the Gregorian calendar structure
+
+const calendar = "buddhist";
+
+for (const { monthCode, daysInMonth } of TemporalHelpers.ISOMonths) {
+  // Test creation with monthCode and day 1
+  const pmd = Temporal.PlainMonthDay.from({ calendar, monthCode, day: 1 });
+  TemporalHelpers.assertPlainMonthDay(pmd, monthCode, 1, `monthCode ${monthCode} should be preserved`);
+
+  // Test with maximum day value for this month (minimum for PlainMonthDay)
+  const pmdMax = Temporal.PlainMonthDay.from({ calendar, monthCode, day: daysInMonth });
+  TemporalHelpers.assertPlainMonthDay(pmdMax, monthCode, daysInMonth, `${monthCode} with day ${daysInMonth} should be valid`);
+
+  // Test constrain overflow
+  const constrained = Temporal.PlainMonthDay.from(
+    { calendar, monthCode, day: daysInMonth + 1 },
+    { overflow: "constrain" }
+  );
+  assert.sameValue(constrained.monthCode, monthCode, `${monthCode} should be preserved with constrain`);
+  assert.sameValue(constrained.day, daysInMonth, `day ${daysInMonth + 1} should be constrained to ${daysInMonth} for ${monthCode}`);
```

Comment:
```suggestion
  TemporalHelpers.assertPlainMonthDay(constrained, monthCode, daysInMonth, `day ${daysInMonth + 1} should be constrained to ${daysInMonth} for ${monthCode}`);
```

Please also look for other places where `TemporalHelpers.assertPlainMonthDay()` can be used instead of individual assertions for `monthCode` and `day`. (At least in the other ISO-like calendars. However, I'd avoid it at least in `chinese` and `dangi` until the issue with reference years is solved.)